cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -109,48 +117,42 @@ static void BM_multibyte_split(benchmark::State& state)
 
   cudaDeviceSynchronize();
 
-  auto source = std::unique_ptr<cudf::io::text::data_chunk_source>(nullptr);
-
-  switch (source_type) {
-    case data_chunk_source_type::file:  //
-      source = cudf::io::text::make_source_from_file(temp_file_name);
-      break;
-    case data_chunk_source_type::host:  //
-      source = cudf::io::text::make_source(host_string);
-      break;
-    case data_chunk_source_type::device:  //
-      source = cudf::io::text::make_source(device_input);
-      break;
-    default: CUDF_FAIL();
-  }
-
-  auto mem_stats_logger = cudf::memory_stats_logger();
-  for (auto _ : state) {
+  auto source = [&] {
+    switch (source_type) {
+      case data_chunk_source_type::file:  //
+        return cudf::io::text::make_source_from_file(temp_file_name);
+      case data_chunk_source_type::host:  //
+        return cudf::io::text::make_source(host_string);
+      case data_chunk_source_type::device:  //
+        return cudf::io::text::make_source(device_input);
+      default: CUDF_FAIL();
+    }
+  }();
+
+  auto mem_stats_logger   = cudf::memory_stats_logger();
+  auto const range_size   = static_cast<int64_t>(device_input.size() * byte_range_factor);
+  auto const range_offset = (device_input.size() - range_size) / 2;
+  cudf::io::text::byte_range_info range{range_offset, range_size};
+  std::unique_ptr<cudf::column> output;
+
+  state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::default_stream_value.value()));
+  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
     try_drop_l3_cache();
-    cuda_event_timer raii(state, true);
-    auto output = cudf::io::text::multibyte_split(*source, delim);
-  }
+    output = cudf::io::text::multibyte_split(*source, delim, range);
+  });
 
-  state.SetBytesProcessed(state.iterations() * device_input.size());
-  state.counters["peak_memory_usage"] = mem_stats_logger.peak_memory_usage();
+  state.add_buffer_size(mem_stats_logger.peak_memory_usage(), "pmu", "Peak Memory Usage");
+  // TODO adapt to consistent naming scheme once established
+  state.add_buffer_size(range_size, "efs", "Encoded file size");
```

Comment:
Can we see some example output? I'm wondering where/how G/s shows up.

Reply:
I added an example output to the PR description. Basically it doesn't, because we don't use the [throughput measurement](https://github.com/NVIDIA/nvbench/blob/main/docs/benchmarks.md#throughput-measurements) functionality. Greg and I talked about this for a bit, and I decided to hold off on adding things here until we have a consistent basis for measuring IO performance. An example for choices we'd need to make: Should the input size be the byte range size, the number of bytes that were read from disk or the size of bytes that ended up in the string column? Also should we count them twice, because we scan first and copy the chars in a second kernel.

Reply:
Nice! I was using `add_element_count` to track peak memory usage but apparently `add_buffer_size` is the better choice.

Reply:
we can calculate throughputs given then existing information, so looks good to me.